### ISSUE TITLE
Generate license metadata API route

### ISSUE THREAD

**Issue by rykci** [lagrangedao/datanft#22]
after the user inputs metadata through the UI, the frontend should send a POST request to the backend, uploading this to MCS and returning the ipfs_url

```JSON
{
    "created_at": "<timestamp>",
    "updated_at": "<timestamp from dataset>",
    "description": "",
    "type": "<dataset, model, or space>",
    "author": "<Name of the entity generating this data>",
    "links": ["Mapping of URL strings for data samples, or links to find out more information."],
    "tags": ["array of keywords"],
    "additionalInformation": {
        "custom_fields": "added by the user"
    }
}
```

note when uploading to MCS, the user can have multiple licenses for a dataset, be careful not to overwrite files in bucket

**Comment by flyworker** [comment]
Do we need MCS add functions to support it?

**Comment by flin-nbai** [comment]
@rykci could you provide more detail about
- `links`, what do you mean by data samples / finding out more information
- `tags`, I don't think we currently assign any tags to datasets. For now this will likely always be empty is that ok?
- `additionalInformation`, what is this additionalInformation section for?

**Comment by flin-nbai** [comment]
> Do we need MCS add functions to support it?

@flyworker I don't think so. Basically frontend lets backend know to generate licensing metadata, backend then creates a json to store this metadata, backend then just uploads the json file like any other MCS file upload. @rykci Could you confirm?

**Comment by rykci** [comment]
Correct the information would be provided by the frontend

> links, what do you mean by data samples / finding out more information

The user can add useful links ex. Website associated with the dataset

> For now this will likely always be empty is that ok?

That's fine

> additionalInformation, what is this additionalInformation section for?

The user would be able to add their own JSON metadata for example:
additionalInformation: {"image": "..png"}

**Comment by flin-nbai** [comment]
How will the user add these extra metadata fields (from the perspective of the API). Will these fields and info be provided as part of the request to the API? And if so, in what format (sent as form, as json, etc)?

**Comment by flyworker** [comment]
on UI it can be an input box
<img width="1083" alt="image" src="https://github.com/lagrangedao/datanft/assets/8363795/3a04c81a-30d0-43bb-b137-717978d2d91f">

**Comment by flyworker** [comment]
https://support.opensea.io/hc/en-us/articles/1500012270982
another interesting information

**Comment by flyworker** [comment]
<img width="551" alt="image" src="https://github.com/lagrangedao/datanft/assets/8363795/16f49c4e-49d6-4a3e-988b-1b17db54b9c1">
